### PR TITLE
Add HTCondor-CE 5 upgrade instructions to OSG-3.5

### PR DESCRIPTION
**Pull request by timtheisen**
Added the HTCondor-CE 5 section. Moved HTCondor 8.8.x section before HTCondor 9.0.0 section. (It's a more logical progression.)

### CODE REVIEW

File: docs/release/updating-to-osg-35.md [hunk 1]
```diff
@@ -149,6 +149,76 @@ To update OSG Configure on your HTCondor-CE, perform the following steps:
...
+!!!tip "Where to find HTCondor-CE 5"
+
+    [HTCondor-CE 5](https://htcondor.github.io/htcondor-ce/v5/releases/), a major
+    version upgrade from HTCondor-CE 4,
+    is available through the [OSG upcoming](release_series.md) repository.
+
+To update HTCondor-CE, perform the following steps:
+
+1.  Merge any `*.rpmnew` files in `/etc/condor-ce/`
+
```
Comment (brianhlin) [inline docs/release/updating-to-osg-35.md:164]: This is probably the most important step to inform folks about: https://htcondor.github.io/htcondor-ce/v5/releases/#new-condor_mapfile-format-and-locations

**Review by brianhlin — APPROVED** [review]
I think the new section can mostly be replaced with a link to the upstream docs https://htcondor.github.io/htcondor-ce/v5/releases/#updating-to-htcondor-ce-5 but if you prefer to keep it this way, you definitely need to address the mapfile changes.

Also, do the release notes link to the specific 9.0.0/5.1.0 upgrade sections? That'd be good in case folks are just looking at the release notes

File: docs/release/updating-to-osg-35.md (hunk 1 above)
Comment (timtheisen) [reply docs/release/updating-to-osg-35.md]: I like this idea. Incorporated it.

**Comment by timtheisen** [comment]
Link checker breakage unrelated and looks transient.

**Pull request merged**